extension/background.js

```diff
@@ -1,112 +1,49 @@
-var windowCount = 'windowCount',
-	incogHistory = 'incogHistory',
-	incogRecent = 'incogRecent';
+if (!chrome.extension.inIncognitoContext) window.close();

```

Comment:
Why are we closing the background window for non-incognito mode?

Reply:
In non-inco, the script is not used at all. So nothing should be "kept". If the script was in the form `(function(){...})()` a `return;` statement would have been enough, but here the popup needs to communicate with the background process (the incognito one) and variables + functions need to be public.

Why did I do that?

IIRC (but I may be wrong), I remember looking at the "Memory" tab of the devtools for the background, and seeing memory used for the environment of the background page (non-inco). Closing the window releases the memory.
I also remember noticing the background page (at least the inco-one) being "swapped" out of memory (I don't remember the word used by google for that event) and woken up when needed. Hence the
```
// pre-wake bg
chrome.runtime.getBackgroundPage(function(){});
```
in the popup window.

I think the 2 led me to unload the non-inco background.

Reply:
Ok, if it reduces memory consumption then we can keep it.